### PR TITLE
Create environment variables plugin.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
PR resolves issue:
https://github.com/eclipse/che/issues/8782

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

File: env-variables-extension/src/browser/env-variables-frontend-module.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { ContainerModule } from "inversify";
+import { IBaseEnvVariablesServer, baseEnvVariablesPath } from '../common/base-env-variables-protocol';
+import { WebSocketConnectionProvider } from '@theia/core/lib/browser';
+
+export default new ContainerModule(bind => {
+    bind(IBaseEnvVariablesServer).toDynamicValue(ctx => {
+        const connection = ctx.container.get(WebSocketConnectionProvider);
+        return connection.createProxy<IBaseEnvVariablesServer>(baseEnvVariablesPath);
+    }).inSingletonScope();
+});
```
Comment (mmorhun) [inline env-variables-extension/src/browser/env-variables-frontend-module.ts]: New line

File: env-variables-extension/src/node/env-variables-backend-module.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        return logger.child({ 'module': 'envs' });
+    }).inSingletonScope().whenTargetNamed("envs");
+
+    bind(ConnectionHandler).toDynamicValue(ctx =>
+        new JsonRpcConnectionHandler(baseEnvVariablesPath, () => {
+            const baseEnvVariablesServer = ctx.container.get<IBaseEnvVariablesServer>(IBaseEnvVariablesServer);
+            return baseEnvVariablesServer;
+        })
+    ).inSingletonScope();
+});
```
Comment (mmorhun) [inline env-variables-extension/src/node/env-variables-backend-module.ts]: Please check all `*.ts` files for new line at the end

**Review by mmorhun — APPROVED** [review]

File: env-variables-extension/src/browser/env-variables-frontend-module.ts (hunk 1 above)
Comment (AndrienkoAleksandr) [reply env-variables-extension/src/browser/env-variables-frontend-module.ts]: Done.

File: env-variables-extension/src/node/env-variables-backend-module.ts (hunk 2 above)
Comment (AndrienkoAleksandr) [reply env-variables-extension/src/node/env-variables-backend-module.ts]: Done

File: README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    cd browser-app
+    yarn watch
+
+Launch `Start Browser Backend` configuration from VS code.
+
+Open http://localhost:3000 in the browser.
+
+## Developing with the Electron example
+
+Start watching of the hello world extension.
```
Comment (azatsarynnyy) [inline README.md]: It's not a `hello world extension` )
FYI: there was a bug in the Yeoman generator for Theia extensions which I've already [fixed](https://github.com/theia-ide/generator-theia-extension/pull/15) a couple of weeks ago.

**Review by akurinnoy — APPROVED** [review]

**Review by ashumilova — APPROVED** [review]

**Pull request merged**